provider/equinix/environ_network.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	if inst == instance.UnknownId {
+		return projectSubnets, nil
+	}
+
+	device, _, err := e.equinixClient.Devices.Get(string(inst), nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var instanceSubnets []network.SubnetInfo
+nextSubnet: //API client limitation since we can't get the actual blocks for individual instance we have to do this
+	for _, psub := range projectSubnets {
+		_, ipnet, err := net.ParseCIDR(psub.CIDR)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for _, n := range device.Network {
+			if ipnet.Contains(net.ParseIP(n.Address)) {
+				instanceSubnets = append(instanceSubnets, psub)
+				continue nextSubnet
+			}
+		}
+	}
+
+	return instanceSubnets, nil
+}
+
+func makeSubnetIDForNetwork(networkName, address string, mask int) (string, string, error) {
+	_, netCIDR, err := net.ParseCIDR(fmt.Sprintf("%s/%d", address, mask))
+	if err != nil {
+		return "", "", errors.Annotatef(err, "calculating CIDR for network %q", networkName)
+	}
+
+	cidr := netCIDR.String()
+	subnetID := fmt.Sprintf("subnet-%s", networkName)
+	return subnetID, cidr, nil
+}
+
+// SupportsSpaces returns whether the current environment supports
+// spaces. The returned error satisfies errors.IsNotSupported(),
+// unless a general API failure occurs.
+func (e *environ) SupportsSpaces(context.ProviderCallContext) (bool, error) {
+	return true, nil
+}
+
+// NetworkInterfaces returns a slice with the network interfaces that
+// correspond to the given instance IDs. If no instances where found, but there
+// was no other error, it will return ErrNoInstances. If some but not all of
+// the instances were found, the returned slice will have some nil slots, and
+// an ErrPartialInstances error will be returned.
+func (e *environ) NetworkInterfaces(ctx context.ProviderCallContext, ids []instance.Id) ([]network.InterfaceInfos, error) {
+	if len(ids) == 0 {
+		return nil, environs.ErrNoInstances
+	}
+	infos := make([]network.InterfaceInfos, len(ids))
+	for i, id := range ids {
+		device, _, err := e.equinixClient.Devices.Get(string(id), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		dev := &equinixDevice{e, device}
+
+		subnets, err := e.Subnets(ctx, id, nil)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		if len(subnets) == 0 {
+			return nil, errors.Trace(fmt.Errorf("Instance does not have subnets"))
+		}
+
+		deviceAddresses, err := dev.Addresses(ctx)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for portIdx, port := range device.NetworkPorts {
+			current := network.InterfaceInfo{
+				DeviceIndex:       portIdx,
+				ProviderId:        network.Id(port.ID),
+				AvailabilityZones: subnets[0].AvailabilityZones,
+				InterfaceType:     network.EthernetDevice,
+				Disabled:          false,
+				NoAutoStart:       false,
+				// Equinix Metal only provides DHCP for the public IPV4
+				ConfigType:    network.ConfigStatic,
+				Origin:        network.OriginProvider,
+				InterfaceName: port.Name,
```

Comment:
@achilleasa We were not sure if `ParentInterfaceName` would be used here for `bond0` / `bond1` when the `ethN` device is bonded.

Reply:
I think you can set ParentInterface="bondX" for the ethN devices if bonded. We typically use this field to indicate that a particular device is bridged.

For example, in the case of lxd, the NICs created for lxd containers will populate ParentInterfaceName with the name of the lxd bridge (e.g. lxdbr0)

@manadart Can you confirm whether it makes sense to populate this field for bonded devices or is it bridge-specific?

Reply:
So I asked around, and you should populate `ParentInterface` name for NICs that are part of the bond. _Generally_, this is not required as the machine agent will collect this information from the machine itself and Juju will merge this information with what the provider reports (the ProviderXXXID fields can obviously only be reported by the provider).

However, as this information is readily available at this point, it's best to be consistent and have the provider report the NIC details with as much fidelity as possible.